arc20.sol

```diff
@@ -0,0 +1,116 @@
+
+pragma solidity >=0.4.22 <0.7.0;
+
+/**
+ * @title Multicoin Wrapper
+ * @dev Wrap Native Asset in ERC20 Standard
+ */
+contract MulticoinERC20Wrapper {
+    
+    address constant balanceAddr = 0x000000000000000000000000000000000000000A;
+    address constant transferAddr = 0x000000000000000000000000000000000000000b;
+    
+    // Track how many tokens are owned by each address.
+    mapping (address => uint256) public balanceOf;
+
+    uint256 public assetID;
+    string public name;
+    string public symbol;
+    uint8 public decimals;
+
+    uint256 public _totalSupply;
+    
+    constructor(string memory _name, string memory _symbol, uint8 _decimals, uint256 _assetID) public {
+        name = _name;
+        symbol = _symbol;
+        decimals = _decimals;
+        assetID = _assetID;
+    }
+    
+    function getBalance(address addr, uint256 aID) public returns (uint256) {
+        (bool success, bytes memory data) = balanceAddr.call(abi.encodePacked(addr, aID));
+        require(success, "getBalance failed");
+        return abi.decode(data, (uint256));
+    }
+    
+    function assetCall(address addr, uint256 aID, uint256 assetAmount, bytes memory callData) public returns (bytes memory) {
+        (bool success, bytes memory data) = transferAddr.call(abi.encodePacked(addr, aID, assetAmount, callData));
+        require(success, "assetCall failed");
+        return data;
+    }
+    
+    function contractBalance() internal returns (uint256) {
+        return getBalance(address(this), assetID);
+    }
```

Comment:
Lets try to be more clear on the term `balance` for if we are talking about the erc20 balance or the underlying ANT balance.